rust/otap-dataflow/crates/otap/src/tls_utils.rs

```diff
@@ -318,6 +374,590 @@ impl ResolvesServerCert for LazyReloadableCertResolver {
     }
 }
 
+/// Internal state for the CA file watcher callback.
+///
+/// # Design: Why Blocking I/O Here is Acceptable
+///
+/// This callback runs in the notify crate's dedicated OS thread, not in the tokio
+/// async runtime. The blocking operations (std::fs::metadata, std::thread::sleep)
+/// only affect the watcher thread, which exists solely to monitor file changes.
+///
+/// Since CA reloads are rare (minutes/hours apart), the performance impact is
+/// negligible. The TLS handshake path remains wait-free - just an atomic pointer load.
+///
+/// Alternative: A channel-based bridge to a tokio worker task would eliminate blocking
+/// entirely, but adds complexity for minimal benefit in this use case.
+struct CaWatcherState {
+    /// The verifier to update on reload
+    inner: Arc<ArcSwap<Arc<dyn ClientCertVerifier>>>,
```

Comment:
This construction Arc<ArcSwap<Arc<dyn T>>>>-- could use more comments!

Reply:
Agree, bit complex - good part is only inner Arc is in the hot-path. Have added the comment for easy understanding.